modules/heat_conduction/doc/content/source/functormaterials/FinEfficiencyFunctorMaterial.md

```diff
@@ -0,0 +1,56 @@
+# FinEfficiencyFunctorMaterial
+
+This [functor material](/FunctorMaterials/index.md) computes a fin efficiency,
+which can be used with [FinEnhancementFactorFunctorMaterial.md], for example.
+
+## Formulation
+
+Under the following assumptions:
+
+- straight fin,
+- uniform cross section, and
+- adiabatic tip,
+
+the fin efficiency $\eta_f$ is the following [!citep](incropera2002):
+
+!equation
+\eta_f = \frac{\text{tanh}(m L)}{m L} \,,
```

Comment:
Hi, this class currently uses the "adiabatic tip" assumption. For an active tip, the approximation that you've listed is appropriate. I just wanted to start simple in the interest of time, figuring others could extend this class as necessary.

Reply:
Sounds good. Thanks!
I assume the addition of other geometry types might require longer time because the Bessel functions are not readily defined in MOOSE (if I am not mistaken).

Reply:
Other geometries are not implemented yet because I have not had the time, and I wanted someone to express interest before investing the time. I'd estimate that I'm tied down for at least a few weeks at the moment.

There are definitely Bessel functions somewhere in MOOSE (probably in the lower level libraries like libMesh). I don't know offhand how to use them. I can poke/ask around if you like - I'm just not familiar.